source/org/zfin/indexer/UiIndexer.java

```diff
@@ -0,0 +1,170 @@
+            System.exit(-1);
+        }
+    }
+
+    protected void index(){
+        List<Entity> output = createInputOutput();
+        cleanUiTables();
+        persistOutput(output);
+    }
+
+    protected abstract List<Entity> inputOutput();
+
+    protected List<Entity> createInputOutput() {
+        IndexerHelper helper = new IndexerHelper();
+        helper.startProcess(getLogPrefix() + " 1. Start Input / Output... ");
+        HibernateUtil.createTransaction();
+        List<Entity> outputList = inputOutput();
+        HibernateUtil.flushAndCommitCurrentSession();
+        helper.addQuickReport(getLogPrefix() + " 1. Finish Input / Output... ", outputList.size());
+        return outputList;
+    }
+
+    protected void persistOutput(List<Entity> resultList) {
+        IndexerHelper helper = new IndexerHelper();
+        helper.startProcess(getLogPrefix() + " 2. Start Persistence... ");
+        saveRecords(createBatch(resultList));
+        helper.addQuickReport(getLogPrefix() + " 2. Finish Persistence... ");
+    }
+
+    private Collection<List<Entity>> createBatch(List<Entity> resultList) {
+        IndexerHelper helper = new IndexerHelper();
+        helper.startProcess();
+        int numberOfBatches = resultList.size() / BATCH_SIZE + 1;
+        AtomicInteger counter = new AtomicInteger();
+        Collection<List<Entity>> batchedList = resultList.stream().collect(Collectors.groupingBy(it -> counter.getAndIncrement() / numberOfBatches)).values();
+        helper.addQuickReport(getLogPrefix() + " Create batches: ");
+        return batchedList;
+    }
+
+    protected abstract void cleanUiTables();
+
+    protected void saveRecords(Collection<List<Entity>> batchedList){
+        HibernateUtil.createTransaction();
+        batchedList.forEach(batch -> {
+            for (Entity entity : batch) {
+                HibernateUtil.currentSession().save(entity);
+            }
+            HibernateUtil.currentSession().flush();
+        });
+        HibernateUtil.flushAndCommitCurrentSession();
+    }
+
+    protected void cleanoutTable(String... tables) {
+        List<String> tableNames = Arrays.stream(tables).map(String::toLowerCase).toList();
+        log.info(getLogPrefix()+" Cleaning out tables: " );
+        tableNames.forEach(table -> {
+            HibernateUtil.createTransaction();
+            getDiseasePageRepository().deleteUiTables(table);
+            HibernateUtil.flushAndCommitCurrentSession();
+        });
+    }
+
+    protected void startTransaction(String message) {
+        indexerHelper.startProcess(message);
+        HibernateUtil.createTransaction();
+    }
+
+    public String calculateRequestDuration(LocalDateTime startTime) {
+        LocalDateTime endTime = LocalDateTime.now();
+        Duration duration = new Duration(startTime, endTime);
+        return duration.toString();
+    }
+
+    public static void main(String[] args) throws NoSuchFieldException {
+
+        args = new String[6];
```

Comment:
why set the args manually here?

Reply:
Removed those lines which were used for debugging purposes onlyl